gm4_mountaineering/data/gm4_mountaineering/functions/climbing_effects/reset_shulker_peek.mcfunction

```diff
@@ -0,0 +1,7 @@
+# reset the peek values of new shulkers and remove the relevant tag
+# @s = player wearing crampons, with using_crampons tag, and facing a climbable block
+# at @s
+# run from add_climb_effects
+
+execute as @e[type=shulker, tag=gm4_mountaineering_climb_assist, tag=gm4_mountaineering_new_shulker] at @s run data merge entity @s {Peek:10}
+tag @e[type=shulker, tag=gm4_mountaineering_climb_assist, tag=gm4_mountaineering_new_shulker] remove gm4_mountaineering_new_shulker
```

Comment:
```suggestion
execute as @e[type=shulker, tag=gm4_mountaineering_climb_assist, tag=gm4_mountaineering_new_shulker] run data merge entity @s {Peek:10}
tag @e[type=shulker] remove gm4_mountaineering_new_shulker
```
minor optimizations: it's actually better not to check tags when removing them. also you don't need `at @s` when just setting the data